salt/modules/virt.py

```diff
@@ -2500,16 +2501,533 @@ def revert_snapshot(name, vm_snapshot=None, cleanup=False, **kwargs):
 
 def _capabilities(conn):
     '''
-    Return connection capabilities
-    It's a huge klutz to parse right,
-    so hide func for now and pass on the XML instead
+    Return connection capabilities as XML.
     '''
     caps = conn.getCapabilities()
     caps = minidom.parseString(caps)
 
     return caps
 
 
+def _get_xml_first_element_by_tag_name(node, name):
+    '''
+    Convenience function getting the first result of getElementsByTagName() or None.
+    '''
+    nodes = node.getElementsByTagName(name)
+    if nodes:
+        return nodes[0]
+    return None
+
+
+def _get_xml_element_text(node):
+    '''
+    Get the text value of an XML element.
+    '''
+    text = ""
+    for child in node.childNodes:
+        if child.nodeType == xml.dom.Node.TEXT_NODE:
+            text = text + child.data
+    return text
```

Comment:
No `+=` on strings, please! At least this:

```python
def _get...
    text = []
    for child...
        if child.nodeType == xml.dom.Node.TEXT_NODE and child.data:
            text.append(child.data)
return [' '].join(text)

Reply:
@isbm Thanks for the review, just pushed changes to address that and use more list/dict comprehensions to get code simpler